Sources/X509/Verifier/TrustRootLoading.swift

```diff
@@ -0,0 +1,99 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftASN1
+
+/// This is a list of root CA file search paths. This list contains paths as validated against several distributions.
+/// If you are attempting to use swift-certificates on a platform that is not covered here and certificate validation is
+/// failing, please open a pull request that adds the appropriate search path.
+private let rootCAFileSearchPaths = [
+    "/etc/ssl/certs/ca-certificates.crt",  // Ubuntu, Debian, Arch, Alpine,
+    "/etc/pki/tls/certs/ca-bundle.crt",  // Fedora
+]
+
+extension CertificateStore {
+    /// A ``CertificateStore`` that includes all root Certificate Authorities (CAs) that
+    /// are installed in the systems trust store.
+    ///
+    /// You can add additional trust roots by ``CertificateStore/inserting(_:)-5sc2d`` them into the returned ``CertificateStore``.
+    ///
+    /// - Note: Access this property as early as possible. It will start loading and parsing of the certificates in the background.
+    /// Accessing this property does **not** block.
+    /// - Warning: This is only supported on Linux and will not find the system trust roots on any other platform.
+    /// On Darwin based platforms (e.g. macOS, iOS) use Security.framework to validate that a certificate chains up to a trusted root CA.
+    /// On other platforms (e.g. Windows) use the platform verifier.
+    public static let systemTrustRoots: CertificateStore = {
+        // access `cachedTrustRootsFuture` to kick off loading on a background thread
+        _ = cachedSystemTrustRootsFuture
+        return CertificateStore(systemTrustStore: true)
+    }()
+
+    static let cachedSystemTrustRootsFuture: Future<[DistinguishedName: [Certificate]], any Error> =
+        DispatchQueue.global(
```

Comment:
Prefer to create a serial dispatch queue for this than directly dispatching to a global queue.